aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/actions/CredentialStoreActions.kt

```diff
@@ -53,11 +69,21 @@ object CredentialStoreActions : StoreActions {
             dispatcher.send(evt)
         }
 
-    override fun loadCredentialStoreAction() =
+    override fun loadCredentialStoreAction(amplifyCredentialType: AmplifyCredentialType, username: String?) =
         Action<CredentialStoreEnvironment>("LoadCredentialStore") { id, dispatcher ->
             logger.verbose("$id Starting execution")
             val evt = try {
-                val credentials = credentialStore.retrieveCredential()
+                val credentials: AmplifyCredential = when (amplifyCredentialType) {
+                    AmplifyCredentialType.AMPLIFY_CREDENTIAL -> {
+                        credentialStore.retrieveCredential()
+                    }
+                    AmplifyCredentialType.DEVICE_METADATA -> {
+                        AmplifyCredential.DeviceData(credentialStore.retrieveDeviceMetadata(username ?: ""))
```

Comment:
is `""` valid username? If yes, what is the behavior?
If no, use `username?.let{}`

Reply:
Removed